Drop after x11/plasma6-libplasma >= 6.1.0 update
Drop after x11-wm/plasma6-kdecoration >= 6.1.0

CMake Warning at CMakeLists.txt:59 (find_package):
  Could not find a configuration file for package "Plasma" that is compatible
  with requested version "6.1".

  The following configuration files were considered but not accepted:

    /usr/local/lib/cmake/Plasma/PlasmaConfig.cmake, version: 6.0.4

CMake Error at CMakeLists.txt:68 (find_package):
  Could not find a configuration file for package "KDecoration2" that is
  compatible with requested version "6.1".

  The following configuration files were considered but not accepted:

    /usr/local/lib/cmake/KDecoration2/KDecoration2Config.cmake, version: 6.0.4

kcms/decoration/declarative-plugin/buttonsmodel.cpp:72:32: error: no member named 'Spacer' in 'KDecoration2::DecorationButtonType'
    case DecorationButtonType::Spacer:
         ~~~~~~~~~~~~~~~~~~~~~~^
kcms/decoration/declarative-plugin/buttonsmodel.cpp:72:32: error: no member named 'Spacer' in 'KDecoration2::DecorationButtonType'
    case DecorationButtonType::Spacer:
         ~~~~~~~~~~~~~~~~~~~~~~^
kcms/decoration/utils.cpp:46:42: error: no member named 'Spacer' in 'KDecoration2::DecorationButtonType'
    {KDecoration2::DecorationButtonType::Spacer, QChar('_')},
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^

--- CMakeLists.txt.orig	2024-06-19 18:47:54 UTC
+++ CMakeLists.txt
@@ -56,7 +56,7 @@ set_package_properties(KF6Kirigami PROPERTIES
     TYPE RUNTIME
 )
 
-find_package(Plasma ${CMAKE_PROJECT_VERSION_MAJOR}.${CMAKE_PROJECT_VERSION_MINOR}
+find_package(Plasma ${CMAKE_PROJECT_VERSION_MAJOR}
   CONFIG
 )
 set_package_properties(Plasma PROPERTIES
@@ -65,7 +65,7 @@ set_package_properties(Plasma PROPERTIES
     TYPE RUNTIME
 )
 
-find_package(KDecoration2 ${CMAKE_PROJECT_VERSION_MAJOR}.${CMAKE_PROJECT_VERSION_MINOR}
+find_package(KDecoration2 ${CMAKE_PROJECT_VERSION_MAJOR}
   CONFIG REQUIRED
 )
 
--- kcms/decoration/declarative-plugin/previewbutton.cpp.orig	2024-06-19 18:47:54 UTC
+++ kcms/decoration/declarative-plugin/previewbutton.cpp
@@ -121,12 +121,7 @@ void PreviewButtonItem::paint(QPainter* painter)
     }
 
     const QRect rect(0, 0, width(), height());
-    if (type() == KDecoration2::DecorationButtonType::Spacer) {
-        static const QIcon icon = QIcon::fromTheme(QStringLiteral("distribute-horizontal"));
-        icon.paint(painter, rect);
-    } else {
         m_button->paint(painter, rect);
-    }
 
     painter->setCompositionMode(QPainter::CompositionMode_SourceAtop);
     painter->fillRect(rect, m_color);
--- kcms/decoration/declarative-plugin/buttonsmodel.cpp.orig	2024-06-19 18:47:54 UTC
+++ kcms/decoration/declarative-plugin/buttonsmodel.cpp
@@ -32,7 +32,6 @@ ButtonsModel::ButtonsModel(QObject* parent)
                        DecorationButtonType::ContextHelp,
                        DecorationButtonType::KeepBelow,
                        DecorationButtonType::KeepAbove,
-                       DecorationButtonType::Spacer,
                    }),
                    parent)
 {
@@ -69,8 +68,6 @@ static QString buttonToName(DecorationButtonType type)
         return i18n("Keep below other windows");
     case DecorationButtonType::KeepAbove:
         return i18n("Keep above other windows");
-    case DecorationButtonType::Spacer:
-        return i18n("Spacer");
     default:
         return QString();
     }
--- kcms/decoration/utils.cpp.orig	2024-06-19 18:47:54 UTC
+++ kcms/decoration/utils.cpp
@@ -43,7 +43,6 @@ const QHash<KDecoration2::DecorationButtonType, QChar>
     {KDecoration2::DecorationButtonType::KeepAbove, QChar('F')},
     {KDecoration2::DecorationButtonType::KeepBelow, QChar('B')},
     {KDecoration2::DecorationButtonType::Shade, QChar('L')},
-    {KDecoration2::DecorationButtonType::Spacer, QChar('_')},
 };
 }
 
